plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java

```diff
@@ -654,7 +649,7 @@ public Collection<Object> createComponents(
             .getClusterSettings()
             .addSettingsUpdateConsumer(MLCommonsSettings.ML_COMMONS_RAG_PIPELINE_FEATURE_ENABLED, it -> ragSearchPipelineEnabled = it);
 
-        LocalClusterIndicesClient localClusterIndicesClient = new LocalClusterIndicesClient(client, xContentRegistry);
+        SdkClient sdkClient = SdkClientFactory.createSdkClient(client, xContentRegistry, settings);
```

Comment:
Cool, so that means we don't need to do ?

```
ModulesBuilder modules = new ModulesBuilder();
modules.add(new SdkClientModule(client, xContentRegistry));

 injector = modules.createInjector();

 // Get the injected SdkClient instance from the injector
 SdkClient sdkClient = injector.getInstance(SdkClient.class);

Reply:
Correct.  That created an injector that was different than the one OpenSearch processes with the list of objects from `createComponents()`.  But the clients still had their runtime class so those would still have been bound to the subclasses, and this really just made a new/different SdkClient available to the various other classes in this method.   We could have just instantiated an sdkClient without jumping through these hoops and gotten the same thing.

With the new code we have a concrete class `SdkClient` that is injected and available by its class key (which is what the separate module was doing) and hides the implementation as an internal field.   Not the most object-oriented way to do things, but it works around the injection headaches.